File: python/SCOOTER/Tests/DW/Manual/flused.py
#==================================================================
#  
#  SCOOTER: Fluid sediment manual case
#  Mexilhoeira Grande, Sab Jul 29 10:22:48 WEST 2017
#  Written by Tordar
#  
#==================================================================

# ipython: run flused

from os import *
import sys
from numpy import *
from scipy.io import *
from pylab import *
sys.path.append ("/home/orodrig/FORdoc/at/Python/")
from readshd import *

rs = 0.0

print("SCOOTER - Fluid sediment manual case")

system("scooter.exe flused")
system("fields.exe flused < flused.flp")

filename = 'flused.shd'
xs = nan
ys = nan
pressure,geometry = readshd(filename,xs,ys)

zs     = geometry["zs"]
rarray = geometry["rarray"]; rarraykm = rarray/1000
zarray = geometry["zarray"]

Dmax = zarray[-1]
rmax = rarray[-1]; rmaxkm = rmax/1000

p = squeeze( pressure )
tl = -20*log10( abs( p ) )
tlmin = min( tl )
tlmax = max( tl )

figure(1)
plot(rarraykm,tl)
xlabel('Range (km)')
ylabel('TL (dB)')
title('SCOOTER - Fluid sediment manual case')
ylim(tlmax,tlmin)
grid(True)

show()

print("done.")

File: python/SCOOTER/Tests/DW/Manual/flused.m
clear all, close all

addpath /home/orodrig/FORdoc/at/Matlab/ReadWrite

disp('SCOOTER - Fluid sediment manual case')

system('cp fields.flp flused.flp');
system('scooter.exe flused');
system('fields.exe flused < flused.flp');

[ PlotTitle, PlotType, freq, atten, Pos, p ] = read_shd( 'flused.shd' );

zs = Pos.s.depth;

rarray = Pos.r.range; rarraykm = rarray/1000;
zarray = Pos.r.depth;

rmax = max( rarray ); rmaxkm = rmax/1000;
zmax = max( zarray );

pressure = squeeze( p );

tl = -20*log10( abs( pressure ) ); 

figure(1)
plot(rarraykm,tl)
xlabel('Range (km)')
ylabel('TL (dB)')
title('SCOOTER - Fluid sediment manual case')
grid on, box on
axis ij

disp('done.')
